soc/arc/snps_nsim/CMakeLists.txt

```diff
@@ -23,6 +23,14 @@ else()
 			       -Xfpus_div -Xfpu_mac -Xfpuda -Xfpus_mpy_slow
 			       -Xfpus_div_slow -Xbitstream -Xtimer0 -Xtimer1)
 
+  zephyr_compile_options_ifdef(CONFIG_SOC_NSIM_EM11D -arcv2em -core3 -Xdiv_rem=radix2
+			       -Xmpy_option=mpyd -Xbitscan -Xswap -Xbarrel_shifter
+			       -Xshift_assist -Xdsp2 -Xdsp_complex -Xxy -Xagu_large
+			       -Xdsp_ctrl=postshift,noguard,convergent -Hfxapi
+			       -Xdsp_divsqrt=radix2 -Xdsp_itu -Xdsp_accshift=full
+			       -Xfpus_div -Xfpu_mac -Xfpuda -Xfpus_mpy_slow
+			       -Xfpus_div_slow -Xbitstream -Xtimer0 -Xtimer1)
```

Comment:
Another issue I see here is that we pass DSP options to generic C flags - so some DSP instructions may be generated while building kernel / application so we may execute DSP instructions in threads which are not marked with 'K_DSP_REGS' flag.

We need somehow distinguish flags for kernel / non-dsp code and dsp code.
Probably we can drop DSP flags from here and just provide DSP flag list in some Cmake variable, so application code can easily add it to the base C flags when building DSP code part.